RecoEgamma/EgammaElectronAlgos/src/GsfElectronAlgo.cc

```diff
@@ -489,6 +497,10 @@ GsfElectronAlgo::EventData GsfElectronAlgo::beginEvent(edm::Event const& event,
                                                 *endcapRecHits,
                                                 &ecalSeveretyLevelAlgo,
                                                 DetId::Ecal),
+      .preselectedTracksForIso03 = tkIsol03Calc_.preselectTracks(*ctfTracks),
+      .preselectedTracksForIso04 = tkIsol04Calc_.preselectTracks(*ctfTracks),
+      .preselectedTracksForIsoHEEP03 = tkIsolHEEP03Calc_.preselectTracks(*ctfTracks),
+      .preselectedTracksForIsoHEEP04 = tkIsolHEEP04Calc_.preselectTracks(*ctfTracks),
```

Comment:
can this be done conditioned on available electrons?
It seems rather heavy to me to process the full track collection 4 times if there are no electrons in the event

Reply:
passing a list of vz of electrons, which is a required cut (if I got the selections right) maybe already good enough